cmd/cartesi-rollups-evm-reader/main.go

```diff
@@ -0,0 +1,193 @@
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/evmreader/service"
+	"github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/internal/repository"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lmittmann/tint"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// Should be overridden during the final release build with ldflags
+	// to contain the actual version number
+	buildVersion = "devel"
+)
+
+const (
+	CMD_NAME                            = "evm-reader"
+	devnetInputBoxDeploymentBlockNumber = uint64(16)
+)
+
+var Cmd = &cobra.Command{
+	Use:   CMD_NAME,
+	Short: "Runs EVM Reader",
+	Long:  `Runs EVM Reader in standalone mode`,
+	Run:   run,
+}
+
+type Config struct {
+	defaultBlock                  string
+	postgresEndpoint              string
+	blockchainHttpEndpoint        string
+	blockchainWsEndpoint          string
+	inputBoxAddress               string
+	inputBoxDeploymentBlockNumber uint64
+}
+
+var verbose bool
+var config Config
+
+func init() {
+
+	Cmd.Flags().StringVarP(&config.defaultBlock,
+		"default-block",
+		"d",
+		"latest",
+		`Default block to be used when fetching new blocks.
+		One of 'latest', 'safe', 'pending', 'finalized'`)
+
+	Cmd.Flags().StringVarP(&config.postgresEndpoint,
+		"postgres-endpoint",
+		"p",
+		"postgres://postgres:password@localhost:5432/postgres",
+		"Postgres endpoint")
+
+	Cmd.Flags().StringVarP(&config.blockchainHttpEndpoint,
+		"blockchain-http-endpoint",
+		"b",
+		"http://localhost:8545",
+		"Blockchain HTTP Endpoint")
+
+	Cmd.Flags().StringVarP(&config.blockchainWsEndpoint,
+		"blockchain-ws-endpoint",
+		"w",
+		"ws://localhost:8545",
+		"Blockchain WS Endpoint")
+
+	Cmd.Flags().StringVarP(&config.inputBoxAddress,
+		"inputbox-address",
+		"i",
+		"0xA1b8EB1F13d8D5Db976a653BbDF8972cfD14691C",
+		"Input Box contract address")
+
+	Cmd.Flags().Uint64VarP(&config.inputBoxDeploymentBlockNumber,
+		"inputbox-block-number",
+		"n",
+		devnetInputBoxDeploymentBlockNumber,
+		"Input Box deployment block number")
+
+	Cmd.Flags().BoolVarP(&verbose,
+		"verbose",
+		"v",
+		false,
+		"enable verbose logging")
+}
+
+func main() {
+	err := Cmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func run(cmd *cobra.Command, args []string) {
+	startTime := time.Now()
+
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	// setup log
+	logLevel := slog.LevelInfo
+	if verbose {
+		logLevel = slog.LevelDebug
+	}
+
+	opts := &tint.Options{
+		Level:      logLevel,
+		AddSource:  logLevel == slog.LevelDebug,
+		TimeFormat: "2006-01-02T15:04:05.000", // RFC3339 with milliseconds and without timezone
+	}
+	handler := tint.NewHandler(os.Stdout, opts)
+	logger := slog.New(handler)
+	slog.SetDefault(logger)
+	slog.Info(
+		"Starting the Cartesi Rollups Node's EVM Reader",
+		"version",
+		buildVersion,
+		"config",
+		config,
+	)
+
+	repository.RunMigrations(fmt.Sprintf("%v?sslmode=disable", config.postgresEndpoint))
```

Comment:
Should not run migrations here. Should assume it was done previously, outside of this scope. You could check if the database is on the expected version.